src/main/java/com/google/codeu/servlets/MessageServlet.java

```diff
@@ -34,53 +34,57 @@
 @WebServlet("/messages")
 public class MessageServlet extends HttpServlet {
 
-  private Datastore datastore;
+	private Datastore datastore;
 
-  @Override
-  public void init() {
-    datastore = new Datastore();
-  }
+	@Override
+		public void init() {
+			datastore = new Datastore();
+		}
 
-  /**
-   * Responds with a JSON representation of {@link Message} data for a specific user. Responds with
-   * an empty array if the user is not provided.
-   */
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+	/**
+	 * Responds with a JSON representation of {@link Message} data for a specific user. Responds with
+	 * an empty array if the user is not provided.
+	 */
+	@Override
+		public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-    response.setContentType("application/json");
+			response.setContentType("application/json");
 
-    String user = request.getParameter("user");
+			String user = request.getParameter("user");
 
-    if (user == null || user.equals("")) {
-      // Request is invalid, return empty array
-      response.getWriter().println("[]");
-      return;
-    }
+			if (user == null || user.equals("")) {
+				// Request is invalid, return empty array
+				response.getWriter().println("[]");
+				return;
+			}
 
-    List<Message> messages = datastore.getMessages(user);
-    Gson gson = new Gson();
-    String json = gson.toJson(messages);
+			List<Message> messages = datastore.getMessages(user);
+			Gson gson = new Gson();
+			String json = gson.toJson(messages);
 
-    response.getWriter().println(json);
-  }
+			response.getWriter().println(json);
+		}
 
-  /** Stores a new {@link Message}. */
-  @Override
-  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+	/** Stores a new {@link Message}. */
+	@Override
+		public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
-    UserService userService = UserServiceFactory.getUserService();
-    if (!userService.isUserLoggedIn()) {
-      response.sendRedirect("/index.html");
-      return;
-    }
+			UserService userService = UserServiceFactory.getUserService();
+			if (!userService.isUserLoggedIn()) {
+				response.sendRedirect("/index.html");
+				return;
+			}
 
-    String user = userService.getCurrentUser().getEmail();
-    String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
+			String user = userService.getCurrentUser().getEmail();
+			String userText = Jsoup.clean(request.getParameter("text"), Whitelist.none());
 
-    Message message = new Message(user, text);
-    datastore.storeMessage(message);
+			String regex = "(https?://\\S+\\.(png|jpg))";
```

Comment:
I did a bit of research on regex (I am certainly not that good at them) and I found this one that might work for more cases for image urls: 
```(http(s?):)([/.\w\s-])*\.(?:jpg|png)```

I'd probably have @edreichua have a look at it since regex is not my strong suit

Reply:
Thanks

Reply:
I tried, this regex didn't work